packages/axios-extension/src/auth/reentrance-ticket/redirect.ts

```diff
@@ -30,35 +29,38 @@ export interface SetupRedirectOptions {
  * @returns a `Redirect` object
  */
 export function setupRedirectHandling({ resolve, reject, timeout, backend, logger }: SetupRedirectOptions): Redirect {
-    const app = express();
-    const server = http.createServer(app);
     const REDIRECT_PATH = '/redirect';
 
+    // eslint-disable-next-line prefer-const
+    let server: http.Server;
+
     const handleTimeout = (): void => {
-        server.close();
+        server?.close();
         reject(new TimeoutError(`Timeout. Did not get a response within ${prettyPrintTimeInMs(timeout)}`));
     };
 
     const timer = setTimeout(handleTimeout, timeout);
-    app.get(REDIRECT_PATH, (req, res) => {
-        if (timer) {
-            clearTimeout(timer);
-        }
-        const reentranceTicket = req.query['reentrance-ticket']?.toString();
-        if (reentranceTicket) {
-            logger.debug('Got reentrance ticket: ' + reentranceTicket);
-            res.set('Content-Type', 'text/html');
-            res.send(Buffer.from(redirectSuccessHtml(backend.logoffUrl())));
-            server.close();
-            resolve({ reentranceTicket, apiUrl: backend.apiHostname() });
-        } else {
-            logger.error('Error getting reentrance ticket');
-            logger.debug(req);
-            res.set('Content-Type', 'text/html');
-            res.status(500);
-            res.send(Buffer.from(redirectErrorHtml()));
-            server.close();
-            reject(new ConnectionError('Error getting reentrance ticket'));
+    server = http.createServer((req, res) => {
+        const reqUrl = new URL(req.url, `http:${req.headers.host}`);
```

Comment:
The second parameter to the `URL()` constructor gets interpolated to a value like `http:localhost:30034/redirect?...`

The code works but the URL standard specifies that the [absolute URL string](https://url.spec.whatwg.org/#absolute-url-string) must have `//` after the scheme. Consider changing this?

Reply:
https://github.com/SAP/open-ux-tools/pull/540/commits/6eca7ee931b046459cc6f27a29d3d4ea4d99d050